src/features/sarcophagi/hooks/useGetSarcophagi.ts

```diff
@@ -0,0 +1,22 @@
+import { ViewStateFacet__factory } from '@sarcophagus-org/sarcophagus-v2-contracts';
+import { useNetworkConfig } from 'lib/config';
+import { Sarcophagus } from 'types';
+import { useContractReads } from 'wagmi';
+
+/**
+ * Makes multiple view calls to the contract retrieving sarcophagi for the sarcoIds provided
+ */
+export function useGetSarcophagi(sarcoIds: string[]): Sarcophagus[] {
+  const networkConfig = useNetworkConfig();
+
+  const { data } = useContractReads({
+    contracts: sarcoIds.map(id => ({
+      address: networkConfig.diamondDeployAddress,
+      abi: ViewStateFacet__factory.abi,
+      functionName: 'getSarcophagus',
+      args: [id],
+    })),
+  });
+
+  return data as Sarcophagus[];
```

Comment:
wagmi is returning a strange object array hybrid here for each element of data 🤔